### DISCUSSION TITLE
Clair scans return 504 Gateway Time-out

### DISCUSSION THREAD

**Discussion by nikolaasdb** [quay/clair#1529] [Q&A] [answered]
Hi,

I'm running Clair in OpenShift and it is used by our Jenkins server for image scanning. 
Most of our builds work properly and Clair scans the layers.
But the moment that one of the layers gets too big, clair returns with an error

``` Exception in thread "main" java.lang.RuntimeException: Request failed: 504 - <html><body><h1>504 Gateway Time-out</h1>
The server didn't respond in time.
</body></html>

at clair.Yair.retry(Yair.java:274)
at clair.Yair.analyseImage(Yair.java:206)
at clair.Yair.scanImage(Yair.java:135)
at clair.Yair.doMain(Yair.java:88)
at clair.Yair.main(Yair.java:69)
```

Do you have any idea what I can do about this? Is there a setting that I haven't found yet that needs to be configured in the config.yaml to change the timeout?

Current Clair version: v4.3.6

Thanks in advance

**Comment by hdonnay** [comment, marked as answer]
Clair doesn't return 504s, this is coming from whatever server/load balancer/proxy you have configured in front of it.
